lib/src/main/scala/spinal/lib/bus/amba4/axis/Axi4StreamSparseCompactor.scala

```diff
@@ -29,51 +29,48 @@ class Axi4StreamSparseCompactor(config: Axi4StreamConfig) extends Component {
 
   io.axis_m << outStage
 
-  def indexOfBoolN(bools: Seq[Boolean], n: Int): Int = {
-    var boolCount = 0
-    for ((b, i) <- bools.zipWithIndex) {
-      if (b) {
-        boolCount += 1
+  def indexOfBoolN(bits: Bits, n: Int): UInt = {
+    val index = UInt(log2Up(widthOf(bits)+1) bit)
+    index := U(widthOf(bits))
+
+    def walkLowBits(width: Int, n: Int): Seq[BigInt] = {
```

Comment:
there is no functional reason to define `walkLowBits` inside `indexOfBoolN` - I looked for a bit which variable of the closure it used...
Also: a short comment about the sequence it generates would have been nice ;-)

Reply:
I originally had the function there to keep from cluttering the higher scope.  I've moved it out and commented its usage and output.